src/plugins/BasebandRIL.js

```diff
@@ -0,0 +1,312 @@
+'use strict';
+
+const OverlayPlugin = require('./util/OverlayPlugin');
+
+/**
+ * Instance Baseband/RIL plugin.
+ * Provides Baseband and RIL informations control.
+ */
+module.exports = class BasebandRIL extends OverlayPlugin {
+    /**
+     * Plugin initialization.
+     *
+     * @param {Object}  instance        Associated instance.
+     * @param {Object}  i18n            Translations keys for the UI.
+     * @param {boolean} basebandEnabled Whether or not baseband control is enabled.
+     */
+    constructor(instance, i18n, basebandEnabled) {
+        super(instance);
+
+        // Reference instance
+        this.instance = instance;
+
+        // Register plugin
+        this.instance.baseband = this;
+        this.i18n = i18n || {};
+
+        this.basebandEnabled = basebandEnabled;
+
+        // Render components
+        this.renderToolbarButton();
+        this.renderWidget();
+
+        // Listen for baseband messages: "<sim/network> <operator/operator_name/imsi_id/phone_number> <value>"
+        this.instance.registerEventCallback('baseband', (message) => {
+            this.HandleBasebandEvent(message);
+        });
+    }
+
+    /**
+     * Handle the baseband channel events.
+     *
+     * @param {String} message the received message.
+     */
+    HandleBasebandEvent(message) {
+        const values = message.split(' ');
+        if (values.length < 3) {
+            return;
+        }
+
+        if (values[0] === 'network' && values[1] === 'operator') {
+            this.networkOperatorMMC.value = values[2];
+        }
+        if (values[0] === 'network' && values[1] === 'operator_name') {
+            this.networkOperatorName.value = values.slice(2).join(' ');
+        }
+        if (values[0] === 'sim' && values[1] === 'operator') {
+            this.simOperatorMMC.value = values[2];
+        }
+        if (values[0] === 'sim' && values[1] === 'operator_name') {
+            this.simOperatorName.value = values.slice(2).join(' ');
+        }
+        if (values[0] === 'sim' && values[1] === 'imsi_id') {
+            this.simMSIN.value = values[2];
+        }
+        if (values[0] === 'sim' && values[1] === 'phone_number') {
+            this.simOperatorPhoneNumber.value = values[2];
+        }
+    }
+
+    /**
+     * Add the button to the player toolbar.
+     */
+    renderToolbarButton() {
+        const toolbars = this.instance.getChildByClass(this.instance.root, 'gm-toolbar');
+        if (!toolbars) {
+            return; // if we don't have toolbar, we can't spawn the widget
+        }
+
+        const toolbar = toolbars.children[0];
+        this.toolbarBtn = document.createElement('li');
+        this.toolbarBtnImage = document.createElement('div');
+        this.toolbarBtnImage.className = 'gm-icon-button gm-sim-button';
+        this.toolbarBtnImage.title = this.i18n.NETWORK_TITLE || 'Baseband';
```

Comment:
Is it normal that we have `NETWORK_TITLE` both in this file and in `Network.js`, for the `toolbarBtnImage.title` ?

Same question for the `title.innerHTML`.

I see in tests, in `baseband.test.js` and `network.test.js` we pass `NETWORK_TITLE: 'TEST NETWORK PLUGIN TITLE'` in both cases. 

I guess I expected something like `BASEBAND_TITLE: 'TEST BASEBAND PLUGIN TITLE'`, for the baseband plugin test, but perhaps I misunderstood something?

Reply:
Yep it should be replaced by BASEBAND